### PR TITLE
Conditional Rendering of Edit Functionality for admins

### LINKED ISSUE #126: Need Conditional Rendering on the Directory & Map Page
**Describe the bug**
If I'm a volunteer, then I shouldn't be able to edit resources. Currently, if I visit the directory or MapView page, I can see the edit button, and I can try to edit these fields, but when I save the edit, I am signed out due to unauthorized request. That's not the best user experience.

**To Reproduce**

- Log in to lah.now.sh
- Click on the Directory tab in the top right corner
- The click on the edit button on any resource
- Click on save or delete

**Expected behavior**
None of the edit flow should be available if you are not an Admin

### PR DESCRIPTION
**Pull request by angadgarg25**
## Status:

:construction: Waiting on #128 to test.

## Description

Conditional rendering of edit functionality so that only admins are able to edit.

Fixes #126 

## Todos
- [ ] Tests

## Screenshots
Volunteer on Map Page: (works on directory page as well)
image
Admin on Map Page: (works on directory page as well)
image

### CODE REVIEW

File: frontend/src/pages/DirectoryView/ResourceCard/index.jsx [hunk 1]
```diff
@@ -40,22 +41,26 @@ const ResourceCard = props => {
         <div className="col col-desc col-desc-collapsed">
           <p>{props.resource.description}</p>
         </div>
-        <div className="col col-edit">
+        {(props.role == roleEnum.ADMIN) && <div className="col col-edit">
```
Comment (josh-byster) [inline frontend/src/pages/DirectoryView/ResourceCard/index.jsx]: Nit: Prefer `===` over `==`

File: frontend/src/utils/enums.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+let roleEnum = {
+  ADMIN: "ADMIN",
+  VOLUNTEER: "VOLUNTEER"
```
Comment (josh-byster) [inline frontend/src/utils/enums.js]: Awesome, good idea - let's add PENDING to this as well so when we merge then @arynh can use this for deciding to render the landing page.

**Comment by josh-byster** [comment]
Merged in #141 which should allow you to test now.

**Comment by angadgarg25** [comment]
> Merged in #141 which should allow you to test now.

works as expected! will go ahead and merge

**Pull request merged**